src/cleanfigure.m

```diff
@@ -296,36 +296,93 @@ function pruneOutsideBox(meta, handle)
+        nextVisible = (dataIsInBox(2:end)   & all(isfinite(data(1:end-1)),2));
+        thisVisible = (dataIsInBox(1:end-1) & all(isfinite(data(2:end)),2));
 
-    [bottomLeft, topLeft, bottomRight, topRight] = corners(xLim, yLim);
+        % Get the corner coordinates
+        [bottomLeft, topLeft, bottomRight, topRight] = corners(xLim, yLim);
 
-    for k = 1:n-1
-        this = data(k  , :);
-        next = data(k+1, :);
+        % Calculate the difference between two consequtive points
+        dataDiff = diff(data);
 
-        % One of the neighbors is inside the box and the other is finite
-        nextVisible = (dataIsInBox(k+1) && all(isfinite(this)));
-        thisVisible = (dataIsInBox(k)   && all(isfinite(next)));
-
-        % Check whether the line connecting this point and the next one
-        % intersects with any of the borders of the drawn axis
-        left   = segmentsIntersect(this, next, bottomLeft , topLeft);
-        right  = segmentsIntersect(this, next, bottomRight, topRight);
-        bottom = segmentsIntersect(this, next, bottomLeft , bottomRight);
-        top    = segmentsIntersect(this, next, topLeft    , topRight);
-
-        % The segment is visible when any of the following hold:
-        %  - this point is visible in the axis and the next is finite
-        %  - this point is finite and the next is visible in the axis
-        %  - the segment connecting this and the next point crosses a border
-        out(k) = thisVisible || nextVisible || left || right || top || bottom;
+        left   = segmentsIntersect(data, dataDiff, bottomLeft , topLeft);
+        right  = segmentsIntersect(data, dataDiff, bottomRight, topRight);
+        bottom = segmentsIntersect(data, dataDiff, bottomLeft , bottomRight);
+        top    = segmentsIntersect(data, dataDiff, topLeft    , topRight);
+
+        % Check the result
+        out = any([thisVisible, nextVisible, left, right, top, bottom],2);
     end
 end
 % =========================================================================
-function out = segmentsIntersect(X1, X2, X3, X4)
+function out = segmentsIntersect(data, dataDiff, X3, X4)
   % Checks whether the segments X1--X2 and X3--X4 intersect.
-  lambda = crossLines(X1, X2, X3, X4);
-  out = all(lambda > 0.0) && all(lambda < 1.0);
+  % Given four points X_k=(x_k,y_k), k\in{1,2,3,4}, and the two lines defined
+  % by those,
+  %
+  %  L1(lambda) = X1 + lambda (X2 - X1)
+  %  L2(lambda) = X3 + lambda (X4 - X3)
+  %
+  % returns the lambda for which they intersect (and Inf if they are parallel).
+  % Technically, one needs to solve the 2x2 equation system
+  %
+  %   x1 + lambda1 (x2-x1)  =  x3 + lambda2 (x4-x3)
+  %   y1 + lambda1 (y2-y1)  =  y3 + lambda2 (y4-y3)
+  %
+  % for lambda1 and lambda2.
+
+  %TODO: why don't we use `\` instead of Cramer's rule?
+
+  n   = size(dataDiff,1);
+  out = false(n,1);
+
+  % Rotational matrix with sign flip 
+  Rotate    = [0, -1; 1, 0 ];   
+  
```

Comment:
I would certainly move it further towards the bottom, i.e. where you actually define the vectorized determinant.